src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -1234,7 +1234,7 @@ static void window_tile_inspector_tool_update(rct_window* w, rct_widgetindex wid
     // Even if Ctrl was pressed, fall back to normal selection when there was nothing under the cursor
     if (clickedElement == nullptr)
```

Comment:
Okay I've been looking into this a bit more. This is still really messy but due to another bug we never actually encounter the bug that would cause screencoords to get clobered with map coords. Thats not a reason to fix the issue though. This function should be taking `screenCoords` not `{mapX, mapY}`. Then we need to get rid of the assignment in L1225-1226. That will then mean we can simplify it so that mapX and mapY mean CoordsXY. I'll fix the bug in `get_map_coordinates_from_pos` myself.

Reply:
Ok, if you want me to do it, let me know

Reply:
I couldn't get away with the whole assignment, as the variables need to be created outside the `if` scope. I did assign just `0`, which I think is what you meant.

I thought about changing it all to `CoordsXY`, but the `get_map_coordinates_from_pos` requires an `int16_t`. I'll work on making sure it is refactored as soon as this is merged.